CRM/Contact/Form/Edit/Address.php

```diff
@@ -165,11 +165,7 @@ public static function buildQuickForm(&$form, $addressBlockCount = NULL, $sharin
       $form->addElement('checkbox', "address[$blockId][use_shared_address]", NULL, ts('Use another contact\'s address'));
 
       // Override the default profile links to add address form
-      $profileLinks = CRM_Core_BAO_UFGroup::getCreateLinks(array(
-          'new_individual',
-          'new_organization',
-          'new_household',
-        ), 'shared_address');
+      $profileLinks = CRM_Contact_BAO_Contact::entityRefCreateLinks() ? CRM_Core_BAO_UFGroup::getCreateLinks('', 'shared_address') : FALSE;
```

Comment:
This cleanup seems to make use of new function but not be related to Campaigns. Should it be in a different MR to make things easier to review and approve?